commands/role.bee.inc

```diff
@@ -0,0 +1,387 @@
+      ),
+    ),
+    'role-add-perm' => array(
+      'description' => bt('Grant specified permission(s) to a role.'),
+      'callback' => 'role_add_permission_bee_callback',
+      'group' => 'roles',
+      'arguments' => array(
+        'permissions' => bt('Permissions'),
+        'role' => bt('Role'),
+      ),
+      'aliases' => array('rap'),
+      'bootstrap' => BEE_BOOTSTRAP_FULL,
+      'examples' => array(
+        'bee role-add-perm  \'post comments\' \'anonymous user\'' => bt("Allow anon users to post comments."),
+        'bee role-add-perm  "\'view own unpublished content\' , \'view any unpublished content\' , \'view revisions\'" \'anonymous user\'' => bt("Grant multiple permissions to the anon users"),
+      ),
+    ),
+    'role-remove-perm' => array(
+      'description' => bt('Remove specified permission(s) from a role.'),
+      'callback' => 'role_remove_permission_bee_callback',
+      'group' => 'roles',
+      'arguments' => array(
+        'permissions' => bt('Permissions'),
+        'role' => bt('Role'),
+      ),
+      'aliases' => array('rrp'),
+      'bootstrap' => BEE_BOOTSTRAP_FULL,
+      'examples' => array(
+        'bee role-remove-perm \'access content\' \'anonymous user\'' => bt("Hide content from anon users."),
+        'bee role-remove-perm  "\'view own unpublished content\' , \'view any unpublished content\' , \'view revisions\'" \'anonymous user\'' => bt("Remove multiple permissions from the anon users"),
+      ),
+    ),
+  );
+}
+
+/**
+ * Command callback: List all roles.
+ */
+function roles_bee_callback($arguments, $options) {
+  $rows = array();
+  $roles = user_roles(TRUE);
+  unset($roles['authenticated']);
+  if (!empty($options['role'])) {
+    // If a role has been specified, check if exists, and return error if
+    // not.
+    $role = strtolower($options['role']);
+    if (!array_key_exists($role, $roles)) {
+      bee_message(bt("The role '!role' doesn't exists!", array(
+        '!role' => $role,
+      )), 'error');
+      return;
+    }
+    $role_array = array(strtolower($options['role']));
+    $permissions = implode('\',\'', user_role_permissions($role_array));
+    if (!empty($permissions)) {
+      $permissions = '\''.$permissions.'\'';
+    }
+    bee_message(bt("The !role role has the following permissions granted: !permissions", array(
+      '!role' => $options['role'],
+      '!permissions' => $permissions,
+    )));
+  }
+  else {
+    $output = array();
+    foreach ($roles as $role => $value) {
+      if (!empty($role)) {
+        $role_array = array($role);
+        $permissions = implode('\' , \'', user_role_permissions($role_array));
+        if (!empty($permissions)) {
+          $permissions = '\''.$permissions.'\'';
+        }
+        $output[] = array(
+          'type' => 'text',
+          'variables' => array(
+            'value' => strtoupper($role) . ': ' . $permissions ,
+          ),
+        );
```

Comment:
What should we output if there are no permissions?
Should we should the role with text saying "No permissions" (using `bt()`)?
Or, should we skip the role and not show anything?

At the moment, it is blank after the role name as there are no permissions, which could be clearer.

Reply:
For me, if it is blank, there are no  permissions. if you wish otherwise let me know.

Reply:
I'll go with your preference here.  I don't have a strong opinion on this.